Framework/Algorithms/test/UnwrapMonitorTest.h

```diff
@@ -0,0 +1,212 @@
+#ifndef MANTID_ALGORITHMS_UNWRAPMONITORTEST_H_
+#define MANTID_ALGORITHMS_UNWRAPMONITORTEST_H_
+
+#include <cxxtest/TestSuite.h>
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Axis.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+#include <string>
+
+#include "MantidAlgorithms/UnwrapMonitor.h"
+
+using namespace Mantid::Algorithms;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Kernel;
+
+class UnwrapMonitorTest : public CxxTest::TestSuite {
+private:
+  void makeFakeWorkspace(std::string wsName) {
+    Workspace2D_sptr test_in =
+        WorkspaceCreationHelper::create2DWorkspaceWithRectangularInstrument(
+            2, 3, 50);
+    test_in->getAxis(0)->setUnit("TOF");
+
+    // Add it to the workspace
+    AnalysisDataService::Instance().add(wsName, test_in);
+  }
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static UnwrapMonitorTest *createSuite() { return new UnwrapMonitorTest(); }
+  static void destroySuite(UnwrapMonitorTest *suite) { delete suite; }
+
+  void testLrefLessThanLd() {
+    // setup
+    std::string name("UnwrapMonitor");
+    this->makeFakeWorkspace(name);
+    Workspace2D_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<Workspace2D>(name);
+    size_t nHist = ws->getNumberHistograms();
```

Comment:
I would avoid using `ws` for the input and then the output later. Calling them something like `inWS` and `outWS` makeso make the tests more readable. Both can be kept then `nHist` is not needed - again I think this makes the test easier to read.